src/thunderbots/software/radio_communication/mrf/dongle.cpp

```diff
@@ -0,0 +1,631 @@
+    //     }
+    //     robots[robot]->handle_message(
+    //         transfer.data() + 1, transfer.size() - 3,
+    //         transfer.data()[transfer.size() - 2],
+    //         transfer.data()[transfer.size() - 1]);
+    // }
+    transfer.submit();
+}
+
+// TODO #222
+void MRFDongle::handle_status(AsyncOperation<void> &)
+{
+    status_transfer.result();
+    estop_state = static_cast<EStopState>(status_transfer.data()[0] & 3U);
+    // if (status_transfer.data()[0U] & 4U)
+    // {
+    //     rx_fcs_fail_message.fire();
+    // }
+    // second_dongle_message.active(status_transfer.data()[0] & 8U);
+    // transmit_queue_full_message.active(status_transfer.data()[0] & 16U);
+    // receive_queue_full_message.active(status_transfer.data()[0] & 32U);
+
+    status_transfer.submit();
+}
+
+
+void MRFDongle::send_camera_packet(std::vector<std::tuple<uint8_t, Point, Angle>> detbots,
+                                   Point ball, uint64_t timestamp)
+{
+    int8_t camera_packet[55] = {0};
+    int8_t mask_vec = 0;  // Assume all robots don't have valid position at the start
+    uint8_t numbots = static_cast<uint8_t>(detbots.size());
+
+    // Initialize pointer to start at location of storing ball data. First 2
+    // bytes are for mask and flag vector
+    int8_t *rptr = &camera_packet[1];
+
+    int16_t ballX = static_cast<int16_t>(ball.x() * 1000.0);
+    int16_t ballY = static_cast<int16_t>(ball.y() * 1000.0);
+
+    *rptr++ = static_cast<int8_t>(ballX);  // Add Ball x position
+    *rptr++ = static_cast<int8_t>(ballX >> 8);
+
+    *rptr++ = static_cast<int8_t>(ballY);  // Add Ball Y position
+    *rptr++ = static_cast<int8_t>(ballY >> 8);
+    struct
+    {
+        bool operator()(std::tuple<uint8_t, Point, Angle> a,
+                        std::tuple<uint8_t, Point, Angle> b) const
+        {
+            return std::get<0>(a) < std::get<0>(b);
+        }
+    } customLess;
+
+    std::sort(detbots.begin(), detbots.end(), customLess);
+
+    // For the number of robot for which data was passed in, assign robot ids to
+    // mask vector and position/angle data to camera packet
+    for (std::size_t i = 0; i < numbots; i++)
+    {
+        uint8_t robotID = std::get<0>(detbots[i]);
+        int16_t robotX  = static_cast<int16_t>((std::get<1>(detbots[i])).x() * 1000);
+        int16_t robotY  = static_cast<int16_t>((std::get<1>(detbots[i])).y() * 1000);
+        int16_t robotT =
+            static_cast<int16_t>((std::get<2>(detbots[i])).toRadians() * 1000);
+
+        mask_vec |= int8_t(0x01 << (robotID));
+        *rptr++ = static_cast<int8_t>(robotX);
+        *rptr++ = static_cast<int8_t>(robotX >> 8);
+        *rptr++ = static_cast<int8_t>(robotY);
+        *rptr++ = static_cast<int8_t>(robotY >> 8);
+        *rptr++ = static_cast<int8_t>(robotT);
+        *rptr++ = static_cast<int8_t>(robotT >> 8);
+
+        /* This was here when I ported the code, no idea what this is for */
```

Comment:
If this seems unnecessary I'd rather just remove it. It's not adding much value by existing in a comment.

Reply:
Yeah, I'll get rid of it. Seems to be working without it anyways.